### PR TITLE
Refit test cases and check wildcards.

### PR DESCRIPTION
**Pull request by abrodersen**
Modified the existing test cases to work with the current code
structure. Added macros to generate the appropriate test cases. Added
code to detect and process wildcard values.

### CODE REVIEW

**Review by jviotti — APPROVED** [review]
Looks great, and the tests pass :+1: Do you think it'd be worth, in a separate PR, to adapt the test runner so we don't have to put compressed JSON in one line in the tests, given its a bit hard to follow right now?

Maybe the different parts of a test case can be detected by using blank lines, and then we can put prettified JSON in there.

**Comment by abrodersen** [comment]
Yeah, I'll look at how the test format can be improved.
